### PR TITLE
[MI-105] Add support for authentication with the api client

### PR DESCRIPTION
**Pull request by miogalang**
Add support for basic authentication and oauth token authentication in the client
Add live tests for checking authentication

/cc @joseconsador @jwswj @mmolina @atroche 

### References
 - Jira link: https://zendesk.atlassian.net/browse/MI-105

### Risks
 - We might not be able to authenticate properly with the client (medium)

### CODE REVIEW

**Comment by joseconsador** [comment]
:+1:

**Comment by miogalang** [comment]
:+1: to @joseconsador 's commit

File: tests/Zendesk/API/LiveTests/BasicTest.php [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    protected $mockedTransactionsContainer = [];
+
+    /**
+     * {@inheritdoc}
+     */
+    public function __construct()
+    {
+        $this->subdomain  = getenv('SUBDOMAIN');
+        $this->username   = getenv('USERNAME');
+        $this->password   = getenv('PASSWORD');
```
Comment (jwswj) [inline tests/Zendesk/API/LiveTests/BasicTest.php]: Do we support password auth? Looks like just token.

**Comment by jwswj** [comment]
Nice :+1:

File: tests/Zendesk/API/LiveTests/BasicTest.php (hunk 1 above)
Comment (miogalang) [inline tests/Zendesk/API/LiveTests/BasicTest.php]: Do we want to support password auth? At the moment we don't but it is easy to add.

Comment (miogalang) [inline tests/Zendesk/API/LiveTests/BasicTest.php]: Yup only token for now. Do we want to add password auth? it should be pretty straightforward. But removing this for now

Comment (jwswj) [inline tests/Zendesk/API/LiveTests/BasicTest.php]: I don't think we should. There is nothing worse than having clear text passwords in code. If it comes up we should be moving people to token based authentication for API clients.

Comment (miogalang) [inline tests/Zendesk/API/LiveTests/BasicTest.php]: I agree 100%

**Pull request merged**